spec/01-unit/04-prefix_handler_spec.lua

```diff
@@ -485,35 +485,71 @@ describe("NGINX conf compiler", function()
     end)
 
     describe("injected NGINX directives", function()
-      it("injects proxy_access_log directive", function()
-        local conf = assert(conf_loader(nil, {
+      it("#test injects proxy_access_log directive", function()
+        local conf, nginx_conf
+        conf = assert(conf_loader(nil, {
           proxy_access_log = "/dev/stdout",
           stream_listen = "0.0.0.0:9100",
           nginx_stream_tcp_nodelay = "on",
         }))
-        local nginx_conf = prefix_handler.compile_kong_conf(conf)
+        nginx_conf = prefix_handler.compile_kong_conf(conf)
         assert.matches("access_log%s/dev/stdout%skong_log_format;", nginx_conf)
-        local nginx_conf = prefix_handler.compile_kong_stream_conf(conf)
+        nginx_conf = prefix_handler.compile_kong_stream_conf(conf)
         assert.matches("access_log%slogs/access.log%sbasic;", nginx_conf)
 
-        local conf = assert(conf_loader(nil, {
+        conf = assert(conf_loader(nil, {
           proxy_access_log = "off",
           stream_listen = "0.0.0.0:9100",
           nginx_stream_tcp_nodelay = "on",
         }))
-        local nginx_conf = prefix_handler.compile_kong_conf(conf)
+        nginx_conf = prefix_handler.compile_kong_conf(conf)
         assert.matches("access_log%soff;", nginx_conf)
-        local nginx_conf = prefix_handler.compile_kong_stream_conf(conf)
+        nginx_conf = prefix_handler.compile_kong_stream_conf(conf)
         assert.matches("access_log%slogs/access.log%sbasic;", nginx_conf)
 
-        local conf = assert(conf_loader(nil, {
+        conf = assert(conf_loader(nil, {
+          proxy_access_log = "/dev/stdout apigw-json",
+          nginx_http_log_format = 'apigw-json "$kong_request_id"',
+          stream_listen = "0.0.0.0:9100",
+          nginx_stream_tcp_nodelay = "on",
+        }))
+        nginx_conf = prefix_handler.compile_kong_conf(conf)
+        assert.matches("access_log%s/dev/stdout%sapigw%-json;", nginx_conf)
+        nginx_conf = prefix_handler.compile_kong_stream_conf(conf)
+        assert.matches("access_log%slogs/access.log%sbasic;", nginx_conf)
+
+        conf = assert(conf_loader(nil, {
+          proxy_access_log = "/dev/stdout apigw-json",
+          nginx_http_log_format = 'not-exist "$kong_request_id"',
+          stream_listen = "0.0.0.0:9100",
+          nginx_stream_tcp_nodelay = "on",
+        }))
+        nginx_conf = prefix_handler.compile_kong_conf(conf)
```

Comment:
this one's compiling an invalid config that would eventually prevent Kong from starting up, right? It is expected, but maybe a little comment above this chunk would help make it more clear.

Reply:
@samugi yep. Previously we print a log or return nil, err.

Let me add a comment.

Reply:
Done.